graph-builder/src/graph.rs

```diff
@@ -88,49 +88,47 @@ impl State {
     }
 }
 
-pub fn run<'a>(opts: &'a config::Options, state: &State) -> ! {
+pub fn run<'a>(settings: &'a config::AppSettings, state: &State) -> ! {
     // Grow-only cache, mapping tag (hashed layers) to optional release metadata.
     let mut cache = HashMap::new();
 
-    let registry = Registry::try_from_str(&opts.registry)
-        .expect(&format!("failed to parse '{}' as Url", &opts.registry));
+    let registry = Registry::try_from_str(&settings.registry)
+        .expect(&format!("failed to parse '{}' as Url", &settings.registry));
 
     // Read the credentials outside the loop to avoid re-reading the file
     let (username, password) =
-        registry::read_credentials(opts.credentials_path.as_ref(), &registry.host)
+        registry::read_credentials(settings.credentials_path.as_ref(), &registry.host)
             .expect("could not read registry credentials");
 
-    let configured_plugins: Vec<Box<plugins::Plugin<plugins::PluginIO>>> =
-        if opts.disable_quay_api_metadata {
-            debug!("Disabling fetching and processing of quay metadata..");
-            vec![]
-        } else {
-            use cincinnati::plugins::internal::{
-                edge_add_remove::EdgeAddRemovePlugin, metadata_fetch_quay::QuayMetadataFetchPlugin,
-                node_remove::NodeRemovePlugin,
-            };
-            use cincinnati::plugins::InternalPluginWrapper;
-
-            // TODO(steveeJ): actually make this vec configurable
-            vec![
-                Box::new(InternalPluginWrapper(
-                    QuayMetadataFetchPlugin::try_new(
-                        opts.repository.clone(),
-                        opts.quay_label_filter.clone(),
-                        opts.quay_manifestref_key.clone(),
-                        opts.quay_api_credentials_path.as_ref(),
-                        opts.quay_api_base.clone(),
-                    )
-                    .expect("could not initialize the QuayMetadataPlugin"),
-                )),
-                Box::new(InternalPluginWrapper(NodeRemovePlugin {
-                    key_prefix: opts.quay_label_filter.clone(),
-                })),
-                Box::new(InternalPluginWrapper(EdgeAddRemovePlugin {
-                    key_prefix: opts.quay_label_filter.clone(),
-                })),
-            ]
+    let configured_plugins: Vec<Box<plugins::Plugin<plugins::PluginIO>>> = {
```

Comment:
The condition for gating the plugins got lost here

Reply:
This is currently unused in our deployment, and #74 takes care of that (plugins not in the config are not enabled). I dropped the CLI flag here.

Reply:
I needed the flag for testing the integration with Satellite. That might come up again anytime soon so I'm in favor to leave something in to gate this.